akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -286,17 +342,31 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
     byteBuffer.clear()
 
     // Write fixed length parts
-    byteBuffer.putInt(header.version)
+    byteBuffer.put(header.version)
+    byteBuffer.put(header.flags)
+    // 1 empty byte slot // TODO avoid having these empty slots
+    // 1 empty byte slot
```

Comment:
sad skipping over not used bytes, please fix during https://github.com/akka/akka/issues/21448